lua/nvim-lsp-installer/servers/solidity_vscode/init.lua

```diff
@@ -0,0 +1,15 @@
+local server = require "nvim-lsp-installer.server"
+local npm = require "nvim-lsp-installer.installers.npm"
+
+return function(name, root_dir)
+    return server.Server:new {
+        name = name,
+        root_dir = root_dir,
+        languages = { "solidity" },
+        homepage = "https://github.com/edag94/vscode-solidity",
+        installer = npm.packages { "solidity-language-server" },
```

Comment:
Who's behind this npm package?

I wonder if we instead should just download the VSCode `.vsix` file from their github releases and unpack it instead? Otherwise, I think it might be worthwhile trying to get this package included in https://github.com/hrsh7th/vscode-langservers-extracted. I suggested the same for the [Spectral LSP](https://github.com/luizcorreia/spectral-language-server/issues/1#issuecomment-981547790), which is another standalone npm package that just wraps a VSCode LSP.

Reply:
@hrsh7th 👋 What are your thoughts on how to deal with all the VSCode LSPs repackaged as npm packages? Do you think we should try to centralize all of them in hrsh7th/vscode-langservers-extracted, or would it make sense to have individual packages maintained by separate individuals? If we were to add more servers to hrsh7th/vscode-langservers-extracted, I think we might have to think about package size and perhaps release a separate npm package per VSCode LSP

Reply:
@williamboman I'm the package publisher.

I forked the [vscode-solidity repo](https://github.com/juanfranblanco/vscode-solidity) and changed the executable to be the server rather than the extension, so that it would fit the npm.executable input (`cmd = { npm.executable(root_dir, "solidity-language-server"), "--stdio" }`), then published it. I also reached out to the owner of the vscode-solidity repo to let him know, and make him a maintainer of the npm package.

Reply:
I am not really sure this overhead (of using the vscode extension) is really worth it. Of course, it is definitely better than nothing, and we already agreed with their author to work together, such that vscode-solidity will be utilizing solc in LSP mode once released, but for nvim I think in the long (!) run does not make too much sense once solc does provide this functionality.

I am writing this here purely as FYI. :-)

Reply:
@christianparpart what is the timeframe for the language server built into solc? I agree that this is the ideal, I was just looking to provide the community with the most viable temp option until the language server is released, with the plan of deprecating it then.

Reply:
@edag94 sorry for the late response. IRL is tight lately. So the initial solc LSP release is imminent (this week (tm)). We start with diagnostics only though, we have way more features implemented already, but want to roll out each in their own cycle for maximum stability. Also: you don't need to defend yourself. I'm actually really glad there is interest in this subject at all. This shows the high need of such a solution and that we should address it in a way that is as close as possible to the upstream language and semantics, which can be best achieved when utilizing the already existing native compiler implementation. :)

Reply:
p.s.: I'm using actually NeoVIM (with coc.nvim) to develop the native LSP ;-)